File: fix_syntax.js
const fs = require('fs');
let content = fs.readFileSync('src/app/rogue-session/start/page.tsx', 'utf8');

// 1. Remove the step 19 block completely
content = content.replace(/\{\s*\/\* --- PAYWALL SECTION --- \*\/\s*\n\s*\{step === 19 && \([\s\S]*?\}\)/, '');

// 2. Decrement steps 20-24
for (let i = 20; i <= 24; i++) {
    content = content.replace(new RegExp(`\\{step === ${i} &&`, 'g'), `{step === ${i - 1} &&`);
}

// 3. Decrement totalSteps
content = content.replace(/const totalSteps = 26;/, 'const totalSteps = 25;');

// 4. Update ResultsOverview to point to /bootcamp
content = content.replace(
    /<Link href=\{isV2 \? "\/v2\/bootcamp" : "\/train\/sales"\} className="inline-block">/,
    `<Link href="/bootcamp" className="inline-block">`
);

content = content.replace(
    /\{isV2 \? "Continue to Day 2" : "Unlock The Full Protocol"\}/,
    `"Continue to Day 2"`
);

// Note: I will intentionally LEAVE PaywallSlide component declaration intact so I don't break JSX at the end of the file. It's just unused code now.

fs.writeFileSync('src/app/rogue-session/start/page.tsx', content);
console.log("Fixed rogue-session cleanly!");

File: fix_syntax.py
import os

files_to_patch = [
    "src/app/rogue-day-4/start/page.tsx",
    "src/app/rogue-day-5/start/page.tsx",
    "src/app/rogue-day-6/start/page.tsx",
    "src/app/rogue-day-7/start/page.tsx",
    "src/app/rogue-day-8/start/page.tsx",
    "src/app/rogue-day-9/start/page.tsx",
    "src/app/rogue-day-10/start/page.tsx",
    "src/app/rogue-day-11/start/page.tsx",
    "src/app/rogue-day-12/start/page.tsx",
    "src/app/rogue-day-13/start/page.tsx",
    "src/app/rogue-day-14/start/page.tsx",
]

for file_path in files_to_patch:
    if not os.path.exists(file_path):
        continue

    with open(file_path, 'r') as f:
        content = f.read()

    # The issue was caused by r"className=\"text-4xl\"" in python
    # which outputs literal backslashes
    content = content.replace('className=\\"text-4xl\\"', 'className="text-4xl"')

    with open(file_path, 'w') as f:
        f.write(content)

    print(f"Fixed {file_path}")

